go/enclave/evm/evm_facade.go

```diff
@@ -186,3 +273,33 @@ func createCleanState(s *state.StateDB, msg *gethcore.Message, ethHeader *types.
 	cleanState.Prepare(chainConfig.Rules(ethHeader.Number, true, 0), msg.From, ethHeader.Coinbase, msg.To, nil, msg.AccessList)
 	return cleanState
 }
+
+func (exec *evmExecutor) chargeBaseFeeOnly(s *state.StateDB, header *types.Header, payer gethcommon.Address, gasUsed uint64) {
+	if gasUsed == 0 || header.BaseFee == nil || header.BaseFee.Sign() == 0 {
+		return
+	}
+	weiBig := new(big.Int).Mul(new(big.Int).SetUint64(gasUsed), header.BaseFee)
+	if weiBig.Sign() <= 0 {
+		return
+	}
+	wei, _ := uint256.FromBig(weiBig)
+	s.SubBalance(payer, wei, tracing.BalanceDecreaseGasBuy)
```

Comment:
are these the right reasons?

Reply:
they are the best from the available ones - the reward one is for tips, but there isn't really a better one for whatever reason.